app/src/main/java/org/fossasia/susi/ai/settings/SettingsActivity.kt

```diff
@@ -0,0 +1,129 @@
+package org.fossasia.susi.ai.settings
+
+import android.content.*
+import android.net.Uri
+import android.os.Bundle
+import android.support.v4.app.FragmentActivity
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.preference.ListPreference
+import android.support.v7.preference.Preference
+import android.support.v7.preference.PreferenceFragmentCompat
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.activities.MainActivity
+import org.fossasia.susi.ai.helper.Constant
+import org.fossasia.susi.ai.login.LoginActivity
+
+/**
+ * Created by mayanktripathi on 07/07/17.
+ */
+
+class SettingsActivity : AppCompatActivity(), ISettingsView {
+
+    var settingsPresenter: ISettingsPresenter? = null
+    var prefs: SharedPreferences? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        prefs = getSharedPreferences(Constant.THEME, Context.MODE_PRIVATE)
+
+        settingsPresenter = SettingsPresenter()
+        settingsPresenter?.onAttach(this, prefs as SharedPreferences)
+
+        if (settingsPresenter?.getThemes() == "Dark") {
+            setTheme(R.style.PreferencesThemeDark)
+        } else {
+            setTheme(R.style.PreferencesThemeLight)
+        }
+
+        setContentView(R.layout.activity_settings)
+    }
+
+    class ChatSettingsFragment : PreferenceFragmentCompat() {
```

Comment:
Move the fragment to a new file "ChatSettingsFragment" and use presenter and model for the fragment. You then won't require presenter and model for activity.